tools/topology/platform/common/ssp.m4

```diff
@@ -27,16 +27,17 @@ define(`SSP_CONFIG',
 $6
 )
 
-dnl SSP_CONFIG_DATA(type, idx, valid bits, mclk_id)
+dnl SSP_CONFIG_DATA(type, idx, valid bits, quirks, mclk_id)
```

Comment:
Why not add the quirks as the last arguments, I think for most SSP config, the mclk_id and quirks will be all 0. You won't need to set them in most cases.

Otherwise you need to modify the m4 files in https://github.com/thesofproject/sof/blob/master/tools/test/topology
They are the only user of mclk_id if you would like to keep the changes now.

Reply:
@xiulipan I missed test topology. The reason i put it in front of mclk_id, I wanted to make only one param optional. If I put it last and make it optional, i think it can be done as you said default value for mclk and quirks are 0. I will try and test also. Let you know if I see more obstacles.

Reply:
@fredoh9 yep, bet to always add new args at the end as this means less changes to the argument parsing (since they are referenced by position and not name).